code/src/savefile.c

```diff
@@ -236,6 +236,47 @@ u8 SaveFile_GetDungeonCount(void) {
     return count;
 }
 
+u8 SaveFile_GetIsSceneDiscovered(u8 sceneNum) {
+    u32 numBits = sizeof(u32) * 8;
+    u32 idx = sceneNum / numBits;
+    u32 bit = 1 << (sceneNum - (idx * numBits));
+    return (gExtSaveData.scenesDiscovered[idx] & bit) != 0;
+}
+
+void SaveFile_SetSceneDiscovered(u8 sceneNum) {
+    u16 numBits = sizeof(u32) * 8;
+    u32 sceneIdx = sceneNum / numBits;
+    u32 sceneBit = 1 << (sceneNum - (sceneIdx * numBits));
+    gExtSaveData.scenesDiscovered[sceneIdx] |= sceneBit;
+}
+
+u8 SaveFile_GetIsEntranceDiscovered(u16 entranceIndex) {
+    u32 numBits = sizeof(u32) * 8;
+    u32 idx = entranceIndex / numBits;
+    u32 bit = 1 << (entranceIndex - (idx * numBits));
+    return (gExtSaveData.entrancesDiscovered[idx] & bit) != 0;
+}
+
+void SaveFile_SetEntranceDiscovered(u16 entranceIndex, u8 isAdult, u8 isNight) {
+    // Most entrances have 4 'slots':
+    // base   - Child Day
+    // base+1 - Child Night
+    // base+2 - Adult Day
+    // base+3 - Adult Night
+    // Based on the values of isAdult and isNight, entranceIndex will be moved to the base index,
+    // and then all four slots will be marked as discovered
+    if (isNight) { entranceIndex -= 1; }
+    if (isAdult) { entranceIndex -= 2; }
+
+    u16 numBits = sizeof(u32) * 8;
+    for (u32 i = 0; i < 4; i++) {
+        u32 entranceIdx = entranceIndex / numBits;
+        u32 entranceBit = 1 << (entranceIndex - (entranceIdx * numBits));
+        gExtSaveData.entrancesDiscovered[entranceIdx] |= entranceBit;
+        entranceIndex++;
+    }
```

Comment:
After poking around some more, since there are much fewer ER overrides than there are entrances, and since the entrances should be unlocked in 'pairs' anyway, what I may do is send a little more data to the patch so that I can figure out which pair, if any, a visited entrance belongs to, and then track completions on the pairs just by index into an array of those. Or, track the entrances like I'm doing (just the base ones, since as you mentioned the game resolves to the base index anyway by the time an area is being loaded), then list each pair, and check if either of the pair's entrances have been visited. This way wastes a small amount of memory, but it's still only around 200 bytes total to keep a list of all 1555 entrances - the string data for all the location names will easily eclipse that on its own.
It is mild spoilers to be able to find out how many different entrance pairs have been randomized with the list, but I do plan to allow the user to turn off ER tracking if they want to.